src/features/activity/ActivityDetailsScreen.tsx

```diff
@@ -0,0 +1,297 @@
+import { ActivityStackParamList } from './activityTypes'
+import BlurActionSheet from '../../components/BlurActionSheet'
+import useHaptic from '../../utils/useHaptic'
+import globalStyles from '../../theme/globalStyles'
+import { DelayedFadeIn } from '../../components/FadeInOut'
+import { useAppStorage } from '../../storage/AppStorageProvider'
+
+type Route = RouteProp<ActivityStackParamList, 'ActivityDetailsScreen'>
+
+const ActivityDetailsScreen = () => {
+  const route = useRoute<Route>()
+  const colors = useColors()
+  const { solanaNetwork: cluster } = useAppStorage()
+  const { t, i18n } = useTranslation()
+  const { triggerNavHaptic } = useHaptic()
+
+  const { transaction } = route.params
+
+  const [optionsOpen, setOptionsOpen] = useState(false)
+  const [selectedAddress, setSelectedAddress] = useState<string>('')
+
+  const dateLabel = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    const date = new Date()
+
+    if (enrichedTx.timestamp) {
+      date.setTime(enrichedTx.timestamp * 1000)
+    }
+
+    if (confirmedSig.blockTime) {
+      date.setTime(confirmedSig.blockTime * 1000)
+    }
+
+    // Format date in DD/MMMM/YYYY format
+    const formattedString = date.toLocaleDateString('en-US', {
+      day: 'numeric',
+      month: 'long',
+      year: 'numeric',
+    })
+
+    const time = `${date.getHours()}:${
+      date.getMinutes() > 9 ? date.getMinutes() : `0${date.getMinutes()}`
+    }`
+
+    return `${formattedString} | ${time}`
+  }, [transaction])
+
+  const activityImage = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err) {
+      return <Error color={colors.error} width={150} height={150} />
+    }
+
+    const { tokenTransfers } = enrichedTx
+
+    if (
+      tokenTransfers &&
+      tokenTransfers[0].tokenMetadata &&
+      tokenTransfers[0].tokenMetadata.json &&
+      tokenTransfers[0].tokenMetadata.json.image
+    ) {
+      return (
+        <ImageBox
+          source={{
+            uri: tokenTransfers[0].tokenMetadata.json.image,
+            cache: 'force-cache',
+          }}
+          width={250}
+          height={250}
+          borderRadius="xxl"
+        />
+      )
+    }
+    return (
+      <CheckmarkFilled color={colors.greenBright500} width={150} height={150} />
+    )
+  }, [colors, transaction])
+
+  const onAddressItemPress = useCallback(
+    (address: string) => () => {
+      setSelectedAddress(address)
+      setOptionsOpen(true)
+    },
+    [],
+  )
+
+  const AccountAddressListItems = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err || !enrichedTx.type) {
+      return null
+    }
+
+    const firstTokenTransfer = enrichedTx.tokenTransfers[0]
```

Comment:
could these ever be empty arrays? You could always use `first` from lodash
```ts
    const firstTokenTransfer = first(enrichedTx.tokenTransfers)
    const firstNativeTransfer = first(enrichedTx.nativeTransfers)
```

Reply:
Yup going to refactor some of this

Reply:
Done